### ISSUE TITLE
OAuth requestToken service doesn't work with Authorization header

### ISSUE THREAD

**Issue by arcao** [opencaching/okapi#512]
Hi Guys,

I'm trying to sign in via OAuth to Opencaching.pl OKAPI service, but it seems OAuth Authorization header params are not used by some validation check. See my attached log:

```text
--> POST https://www.opencaching.pl/okapi/services/oauth/request_token
Content-Type: application/x-www-form-urlencoded
Content-Length: 0
Authorization: OAuth oauth_nonce="1480808688", oauth_signature="s3EFZMQnHYGg9JaW15sKfihIUTY%3D", oauth_callback="x-oc4l%3A%2F%2Foauth.callback%2F", oauth_consumer_key="9w4tqST4TUNrGYQKnkgQ", oauth_timestamp="1505651506", oauth_signature_method="HMAC-SHA1", oauth_version="1.0"
--> END POST (0-byte body)

<-- 400 Bad Request https://opencaching.pl/okapi/services/oauth/request_token (480ms)
Date: Sun, 17 Sep 2017 12:32:11 GMT
Server: Apache/2.4.20 (Ubuntu)
Access-Control-Allow-Origin: *
Content-Length: 242
Connection: close
Content-Type: application/json; charset=utf-8
{"error":{"developer_message":"This method requires OAuth signature (Level 2 Authentication). You didn't sign your request.","reason_stack":["bad_request"],"status":400,"more_info":"https:\/\/opencaching.pl\/okapi\/introduction.html#errors"}}
<-- END HTTP (242-byte body)
```

I tried to switch to send OAuth params via query string, but this returns me invalid signature error,
 maybe bug in ScribeJava library. Anyway I have tested that ScribeJava works OK with Authorization header on another project.

**Comment by wrygiel** [comment]
Hmm, might be caused by the [recent changes](https://github.com/opencaching/okapi/issues/475) to OAuth handling code. Thanks, we will look into that.

**Comment by arcao** [comment]
I tried to test with latest opencaching/opencaching-pl@6b86831da1d7e674d5147c426a3248a72820e866 on VMWare Workstation and it works OK. Is there any difference between this repo and implementation in opencaching-pl?

**Comment by wrygiel** [comment]
No, the code should be the same. Have you tried the VMware option with HTTPS, or HTTP?

I didn't have the time to debug this yet. How urgent do you think it is?

**Comment by wrygiel** [comment]
I have reviewed the changes from 195bcb81a819d7c222153c62f6a9c672184b926a and a73ba55af3f2c15ec419130cd61e3d500600ce19 (the commits I suspected might be faulty), but couldn't find there anything which could lead to your problem.

I have never tested OKAPI's OAuth via the `Authorization` header myself. Perhaps it never had worked...?

**Comment by arcao** [comment]
I've do test with unsecured http and there it works OK. So the problem is related to OAuth via https protocol only. Anyway thanks for help.

**Comment by wrygiel** [comment]
If I understand correctly, the problem surfaces only when both HTTPS and OAuth `Authorization` header is used. Weird, but possible. Perhaps our OAuth library (quite old now) isn't HTTPS-ready and needs to be upgraded.

In other words, this bug doesn't break *existing* apps, but it still makes harder to use OKAPI for new apps.

**Comment by arcao** [comment]
There are two bugs which prevent to use OkApi on HTTPS environment:

1. I don't know how SSL support is implemented on opencaching.pl server (native or via reverse proxy) but it seems, the HTTPS is not detected correctly in PHP. I think that while creating base string for signature, `http` protocol is used for resource url instead of `https`. Because I finally get `oauth_signature` incorrect error. This breaks OAuth via GET query and maybe also via `Authorization` header.

2. Authorization header is not recognized on HTTPS environment. It seems PHP doesn't get it.

With unsecured HTTP both variant with OAuth `Authorization` header and GET query works OK.

**Comment by following5** [comment]
@arcao 
Opencaching.pl recently moved to a new server. Could you give it another try?

**Comment by sdennler** [comment]
@following5 
I do have a example where it works with opencaching.pl but fails with opencaching.de because of the `Authorization` header is not recognized: https://github.com/sdennler/okapi-auth-example

**Comment by sdennler** [comment]
@following5
Found the issue: The header is not forwarded when redirecting from `opencaching.de` to `www.opencaching.de`. Using `www.opencaching.de` directly works 👍
